bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -977,26 +967,8 @@ protected void addThingHandlerFactory(ThingHandlerFactory thingHandlerFactory) {
 
     private void registerAndInitializeHandler(final Thing thing, final ThingHandlerFactory thingHandlerFactory) {
         if (thingHandlerFactory != null) {
-            try {
-                SafeMethodCaller.call(new SafeMethodCaller.ActionWithException<Void>() {
-                    @Override
-                    public Void call() throws Exception {
-                        registerHandler(thing, thingHandlerFactory);
-                        initializeHandler(thing);
-                        return null;
-                    }
-                });
-            } catch (TimeoutException e) {
-                logger.warn("Registering a handler for thing '{}' takes more than {}ms.", thing.getUID(),
-                        SafeMethodCaller.DEFAULT_TIMEOUT);
-            } catch (Exception ex) {
-                ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
```

Comment:
both `registerHandler(...)` and `initializeHandler(...)` do exception handling themselves, also setting the thing status accordingly. What are you missing here?